src/mjolnir/pbfgraphparser.cc

```diff
@@ -1927,8 +1927,10 @@ struct graph_callback : public OSMPBF::Callback {
         // node
         n.set_state_iso_index(osmdata_.node_names.index(tag.second));
         ++osmdata_.node_name_count;
+      } else if (tag.first == "crossing" && tag.second == "traffic_signals") {
+        n.set_traffic_signal(true);
       } else if (tag.first == "highway") {
-        n.set_traffic_signal(tag.second == "traffic_signals");
+        n.set_traffic_signal(n.traffic_signal() || tag.second == "traffic_signals");
```

Comment:
this has the potential to make the alogrithm think that we have to stop multiple times in a row in quick succession when a car traffic light and a cross walk light are next to each other. we might want to move pedestrian crossing to be a new node type or use the spare bit on node